File: FRONTEND/src/types/video.ts
export type VideoCreate = {
  course_id: number;
  title: string;
  content: string;
  url: string | null;
  file: string | null;
  descarga: string | null;
  seccion: number;
  title_accordion: string | null;
};

export type VideoUpdate = {
  course_id: number | null;
  title: string | null;
  content: string | null;
  url: string | null;
  file: string | null;
  descarga: string | null;
  seccion: number | null;
  title_accordion: string | null;
};

export type Video = VideoCreate & {
  id: number;
  user_id: number;
  deleted_at: string | null;
  created_at: string | null;
  updated_at: string | null;
};

File: BACKEND/app/schemas/video.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class VideoCreate(BaseModel):
    course_id: int
    title: str = Field(min_length=1, max_length=255)
    content: str = Field(min_length=1)
    url: str | None = Field(default=None, max_length=255)
    file: str | None = None
    descarga: str | None = None
    seccion: int
    title_accordion: str | None = Field(default=None, max_length=255)


class VideoUpdate(BaseModel):
    course_id: int | None = None
    title: str | None = Field(default=None, min_length=1, max_length=255)
    content: str | None = Field(default=None, min_length=1)
    url: str | None = Field(default=None, max_length=255)
    file: str | None = None
    descarga: str | None = None
    seccion: int | None = None
    title_accordion: str | None = Field(default=None, max_length=255)


class VideoPublic(VideoCreate):
    id: int
    user_id: int
    deleted_at: datetime | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)
